textworld/generator/text_generation.py

```diff
@@ -437,6 +444,9 @@ def assign_description_to_quest(quest, game, grammar):
             quest_tag = quest_tag.replace("(list_of_actions)", actions_desc.strip())
 
         quest_desc = grammar.expand(quest_tag)
+        quest_desc = re.sub(r"([!.]) ([a-z])",
```

Comment:
For good measure, we can generalize it to
```suggestion
        quest_desc = re.sub(r"(^|(?<=[?!.]))\s*([a-z])", lambda m: m.group(0).upper(), quest_desc)
```